tests/test_api.py

```diff
@@ -0,0 +1,133 @@
+from typing import Any, Generator
+
+from fastapi.testclient import TestClient
+import pytest
+from sqlalchemy import StaticPool, create_engine
+from sqlalchemy.orm import Session, sessionmaker
+
+from src.core.db import Base, get_db
+from src.main import app
+from src.models import Student as StudentModel
+from src.schemas import StudentBase as StudentSchema
+
+client = TestClient(app)
+
+
+SQLALCHEMY_DATABASE_URL = "sqlite:///:memory:"
+engine = create_engine(
+    SQLALCHEMY_DATABASE_URL,
+    connect_args={
+        "check_same_thread": False,
+    },
+    poolclass=StaticPool,
+)
+TestingSessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+
+def override_get_db() -> Generator[Session, None, None]:
+    db = TestingSessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+app.dependency_overrides[get_db] = override_get_db
+
+
+@pytest.fixture
+def mock_student_model() -> StudentModel:
+    return StudentModel(name="John Doe", age=21, email="john@example.com", major="Compsci")
+
+
+@pytest.fixture
+def mock_student_schema() -> StudentSchema:
+    return StudentSchema(name="Jane Doe", age=21, email="jane@example.com", major="Compsci")
+
+
+@pytest.fixture(autouse=True)
+def setup_and_teardown(mock_student_model: StudentModel) -> Generator[Any, Any, Any]:
+    Base.metadata.create_all(bind=engine)
+
+    db = TestingSessionLocal()
+    db.add(mock_student_model)
+    db.commit()
+    db.refresh(mock_student_model)
+
+    yield
+    Base.metadata.drop_all(bind=engine)
+
+
+def test_read_root() -> None:
+    response = client.get("/")
+    assert response.status_code == 200
+    assert response.json() == {"message": "Hello World"}
+
+
+def test_read_students() -> None:
+    response = client.get("/students/")
+    assert response.status_code == 200
+    assert response.json() == [
+        {"id": 1, "name": "John Doe", "age": 21, "email": "john@example.com", "major": "Compsci"}
+    ]
+
+
+def test_create_student(mock_student_schema: StudentSchema) -> None:
+    response = client.post(
+        "/students/",
+        json={**mock_student_schema.model_dump(), "age": 22},
```

Comment:
Tutaj pytanie z ciekawości: czemu nadpisujesz age w słowniku? :D

Reply:
To była pozostałość wcześniejszego "planu", przed chwilą to zauważyłem jak przeglądałem kod i usunąłem. Teraz mam "klonowanego" studenta w fixture (dla modelu z SQLAlchemy i schematu z Pydantica) i nadpisuję email, żeby nie wyrzucał błędu, że ma konflikt.